lib/cfg-lexer.c

```diff
@@ -946,18 +946,22 @@ cfg_lexer_lex(CfgLexer *self, YYSTYPE *yylval, YYLTYPE *yylloc)
           free(yylval->cptr);
           cfg_args_unref(args);
 
-          cfg_block_generator_format_name(gen, buf, sizeof(buf));
-
-          if (gen->suppress_backticks)
-            success = cfg_lexer_include_buffer_without_backtick_substitution(self, buf, result->str, result->len);
-          else
-            success = cfg_lexer_include_buffer(self, buf, result->str, result->len);
-          g_string_free(result, TRUE);
-
           if (success)
             {
+              cfg_block_generator_format_name(gen, buf, sizeof(buf));
+
+              if (gen->suppress_backticks)
+                success = cfg_lexer_include_buffer_without_backtick_substitution(self, buf, result->str, result->len);
+              else
+                success = cfg_lexer_include_buffer(self, buf, result->str, result->len);
+              g_string_free(result, TRUE);
+
               goto relex;
```

Comment:
The original code conditionally goto-ed to relex, when `g_lexer_include_buffer_without_backtick_substitution` and `cfg_lexer_include_buffer` returned successful. Can we skip to check the return values now?